File: src/main/java/com/raihanorium/springreact/model/Company.java
package com.raihanorium.springreact.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;

@Entity
@Data
public class Company {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;
    private String contactPerson;
    private String contactNo;
    private String officeAddress;
}

File: src/main/frontend/src/model/Company.ts
export class Company {

  public static from(formData: FormData): Company {
    return new Company(
        formData.get("id") as number | null,
        formData.get("companyName") as string | null,
        formData.get("contactPerson") as string | null,
        formData.get("contactNo") as string | null,
        formData.get("officeAddress") as string | null,
    );
  }

  public toObject() {
    return {
      name: this.name,
      contactPerson: this.contactPerson,
      contactNo: this.contactNo,
      officeAddress: this.officeAddress,
    };
  }

  public constructor(
      public id: number | null,
      public name: string | null,
      public contactPerson: string | null,
      public contactNo: string | null,
      public officeAddress: string | null,
  ) {
  }
}